frontend/js/src/artist-similarity/src/artist-search/SearchBox.tsx

```diff
@@ -0,0 +1,154 @@
+
+    // Hide the dropdown when the user clicks outside of it
+    const toggleDropdown = () => setTimeout(() => {
+        const dropdown = document.getElementById("search-dropdown");
+        dropdown?.style.display === "flex" ? dropdown.style.display = "none" : dropdown!.style.display = "flex";
+    }, dropdownDelay);
+
+    const increment = () => {
+        let input = document.getElementById("graph-size-input-number") as HTMLInputElement;
+        input.stepUp();
+        // Dispatch change event to trigger onChange, so the custom buttons also trigger the event
+        input.dispatchEvent(new Event('change', { bubbles: true }));
+        
+    }
+    const decrement = () => {
+        let input = document.getElementById("graph-size-input-number") as HTMLInputElement;
+        input.stepDown();
+        // Dispatch change event to trigger onChange, so the custom buttons also trigger the event
+        input.dispatchEvent(new Event('change', { bubbles: true }));
+    }
+    return (
+        <form
+        className="search-box"
+        autoComplete="off"
+        >
+            <div
+            className="artist-input"
+            onFocus={toggleDropdown}
+            onBlur={toggleDropdown}
+            >
+                <div
+                className="artist-input-box"
+                >
+                    <input 
+                    id="artist-input-name" 
+                    type="search"
+                    name="artist_mbid" 
+                    placeholder="Artist name" 
+                    onChange={e => setSearchQuery(e.target.value)}
+                    value={searchQuery}
+                />
+                    <button 
+                    id="artist-input-icon"
+                    type="button"
+                    >
+                        <FontAwesomeIcon 
+                        icon={faMagnifyingGlass} 
+                        color="white"
+                        />   
+                    </button>
+                </div>
+                <SearchDropdown 
+                searchResults={searchResults} 
+                onArtistChange={setArtistMbid} 
+                id={"search-dropdown"}
+                />
+            </div>
+            <div
+            className="graph-size-input"
+            >
+            <label 
+            id="graph-size-input-label"
+            >
+                Web size:
+            </label>
+            <button
+            id="graph-size-input-minus"
+            type="button"
+            onClick={decrement}
+            >
+                <FontAwesomeIcon 
+                icon={faMinus} 
+                color="white"
+                />
+            </button>
+            <input 
+            id="graph-size-input-number" 
+            type="number" 
+            name="similarArtistsLimit" 
+            placeholder="Graph size" 
+            min="1" max="25"  
+            onChange={handlesimilarArtistsLimitChange} 
+            defaultValue={similarArtistsLimit}
```

Comment:
```suggestion
            value={similarArtistsLimit}
```
Could this be the cause of your need for the increment and decrement functions to trigger the change event?

Reply:
Fixed